rewrite-maven/src/main/java/org/openrewrite/maven/RemoveRepository.java

```diff
@@ -0,0 +1,140 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.maven;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Option;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.xml.AddToTagVisitor;
+import org.openrewrite.xml.ChangeTagValueVisitor;
+import org.openrewrite.xml.RemoveContentVisitor;
+import org.openrewrite.xml.XPathMatcher;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.Objects;
+import java.util.Optional;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RemoveRepository extends Recipe {
+
+    private static final XPathMatcher REPOS_MATCHER = new XPathMatcher("/project/repositories/repository");
+
+    @Option(required = false, description = "Repository id")
+    @Nullable
+    private String id;
+
+    @Option(description = "Repository URL")
+    private String url;
+
+    @Option(required = false, description = "Repository name")
+    @Nullable
+    private String repoName;
+
+    @Option(required = false, description = "Repository layout")
+    @Nullable
+    private String layout;
+
+    @Option(required = false, description = "Snapshots from the repository are available")
+    @Nullable
+    private Boolean snapshotsEnabled;
+
+    @Option(required = false, description = "Snapshots checksum policy")
+    @Nullable
+    private String snapshotsChecksumPolicy;
+
+    @Option(required = false, description = "Snapshots update policy policy")
+    @Nullable
+    private String snapshotsUpdatePolicy;
+
+    @Option(required = false, description = "Releases from the repository are available")
+    @Nullable
+    private Boolean releasesEnabled;
+
+    @Option(required = false, description = "Releases checksum policy")
+    @Nullable
+    private String releasesChecksumPolicy;
+
+    @Option(required = false, description = "Releases update policy")
+    @Nullable
+    private String releasesUpdatePolicy;
+
+    @Override
+    public String getDisplayName() {
+        return "Remove repository";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Removes a matching Maven repository.";
+    }
+
+    @Override
+    protected TreeVisitor<?, ExecutionContext> getVisitor() {
+        return new MavenIsoVisitor<ExecutionContext>() {
+            @Override
+            public Xml.Tag visitTag(Xml.Tag tag, ExecutionContext ctx) {
+                Xml.Tag repo = super.visitTag(tag, ctx);
+
+                if (REPOS_MATCHER.matches(getCursor())) {
+                    if (id == null && (isReleasesEqual(repo) && isSnapshotsEqual(repo)) &&
+                                        url.equals(repo.getChildValue("url").orElse(null))) {
```

Comment:
@jkschneider what if I want to match something like the regular expression: `http[s]+://maven.glassfish.org/?.*`
Is there a better way we could do matching here that allows for more generic removals like this?